### PR TITLE
Add strictMode config option for non-fuzzy license checking

### PR DESCRIPTION
**Pull request by palmerj3**
Example: by default simply adding "BSD" to the "allowedLicenses" matches "BSD", "BSD*", "BSD-3-Clause", "BSD-4-Clause", etc.

strictMode does a non-fuzzy check so you would, if you choose, be able to allow for BSD-3-Clause in "allowedLicenses" but not BSD-4-Clause.

### CODE REVIEW

**Comment by hzoo** [comment]
I guess we should setup some tests after this 😛

**Comment by palmerj3** [comment]
@hzoo not a bad idea :)  Let me know how you want to proceed. I can either add some in a separate PR or along with this one.

**Comment by palmerj3** [comment]
@hzoo I modified the code a bit so it's more testable (at least the parts I wanted to test) and created some jest tests for strictMode.

**Comment by hzoo** [comment]
This is great! dono about 0.10 I guess I forgot I didn't mean to support that (would be breaking change) but will just drop it.

Also realized we can't use jest (via PATENTs/license) lol.. irony.

You cool with switching to mocha/ava/jasmine? I can do it later too.

**Comment by palmerj3** [comment]
@hzoo sure I can switch it to mocha or ava. If you get to it before me then feel free - other wise I'll ping you when it's done :)

**Comment by palmerj3** [comment]
@hzoo ok I converted it to ava instead of jest

File: package.json [hunk 1]
```diff
@@ -13,11 +13,14 @@
   },
   "scripts": {
     "check-license": "./bin/license-to-fail.js ./example-config.js",
-    "test": "npm run check-license"
+    "test": "ava && npm run check-license"
```
Comment (hzoo) [inline package.json]: Cool I believe we need to modify `files` now for `index.js` since it was moved

Comment (palmerj3) [inline package.json]: @hzoo all set

**Comment by palmerj3** [comment]
@hzoo let me know if there is anything else you need me to change. Thanks!

**Comment by hzoo** [comment]
Cool will plan to merge/release tomorrow

**Comment by palmerj3** [comment]
@hzoo think we'll merge this soon?  :)

**Comment by hzoo** [comment]
Ah totally forgot about this and on break so haven't being doing much oss, sorry

**Comment by hzoo** [comment]
Can do it when I get my on computer later

File: package.json [hunk 2]
```diff
@@ -6,18 +6,22 @@
   "files": [
     "bin",
     "index.js",
```
Comment (hzoo) [inline package.json]: we can remove `index.js`

File: bin/license-to-fail.js [hunk 3]
```diff
@@ -43,4 +43,4 @@ config.__currentPackage = {
   optionalDependencies: packageJson.optionalDependencies || []
 };
 
-require("../index")(config);
+require("../index")(config).check();
```
Comment (hzoo) [inline bin/license-to-fail.js]: Does this need to be `../src/check-licenses.js`?

**Comment by hzoo** [comment]
sorry bout that @palmerj3, taking a break from oss stuff this holiday but should of did this earlier

**Pull request merged**